src/extension/providers/commonProvider/index.ts

```diff
@@ -0,0 +1,43 @@
+import { EventEmitter } from "events";
+import { postMessage } from "~/extension/providers/postMessage";
+import { PromiseQueue } from "~/extension/providers/promiseQueue";
+
+export default class CommonProvider {
+  enabled: boolean;
+  private _eventEmitter: EventEmitter;
+  private _queue: PromiseQueue;
+
+  constructor() {
```

Comment:
Can we pass the scope to the base class? That would allow us to get rid of the first parameter for every `execute()` call from deriving classes.